servers/src/main/java/tachyon/worker/block/evictor/LRFUEvictor.java

```diff
@@ -0,0 +1,345 @@
+      List<Long> toEvict = new ArrayList<Long>();
+      EvictionPlan plan = new EvictionPlan(toMove, toEvict);
+      StorageDirView candidateDir = cascadingEvict(bytesToBeAvailable, location, plan, sortedCRF);
+
+      if (candidateDir == null) {
+        return null;
+      }
+
+      return plan;
+    }
+  }
+  
+  /**
+   * Sort all blocks in ascending order of CRF
+   * 
+   * @return the sorted CRF of all blocks
+   */
+  private List<Map.Entry<Long, Double>> getSortedCRF() {
+    List<Map.Entry<Long, Double>> sortedCRF = 
+        new ArrayList<Map.Entry<Long, Double>>(mBlockIdToCRFValue.entrySet());
+    Collections.sort(sortedCRF, new Comparator<Map.Entry<Long, Double>>() {
+      public int compare(Entry<Long, Double> o1, Entry<Long, Double> o2) {
+        double res = o1.getValue() - o2.getValue();
+        if (res < 0) {
+          return -1;
+        } else if (res > 0) {
+          return 1;
+        } else {
+          return 0;
+        }
+      }
+    });
+    return sortedCRF;
+  }
+
+  /**
+   * Calculate function F(t) = pow (1.0 / {@link #mAttenuationFactor}, t * {@link #mStepFactor})
+   * 
+   * @param logicTimeInterval time interval since that access to current
+   * @return Function value of F(t)
+   */
+  private double calculateFunction(long logicTimeInterval) {
+    return 1.0 * Math.pow(1.0 / mAttenuationFactor, 1.0 * logicTimeInterval * mStepFactor);
+  }
+
+  /**
+   * This function is used to update CRF of all the blocks according to current logic time. When
+   * some block is accessed in some time, only CRF of that block itself will be updated to current
+   * time, other blocks who are not accessed recently will only be updated until 
+   * {@link #freeSpaceWithView(long, BlockStoreLocation, BlockMetadataManagerView)} is called 
+   * because blocks need to be sorted in the increasing order of CRF. When this function is called,
+   * {@link #mBlockIdToLastAccessTime} and {@link #mBlockIdToCRFValue} need to be locked in case
+   * of the changing of values.
+   */
+  private void updateCRFValue() {
+    long currentLogicTime = mLogicTimeCount.get();
+    for (Iterator<Map.Entry<Long, Double>> it = mBlockIdToCRFValue.entrySet().iterator(); it
+        .hasNext();) {
+      Map.Entry<Long, Double> entry = it.next();
+      long blockId = entry.getKey();
+      double crfValue = entry.getValue();
+      mBlockIdToCRFValue.put(blockId, crfValue
+          * calculateFunction(currentLogicTime - mBlockIdToLastAccessTime.get(blockId)));
+      mBlockIdToLastAccessTime.put(blockId, currentLogicTime);
+    }
+  }
+
+  @Override
+  public void onAccessBlock(long userId, long blockId) {
+    synchronized (mBlockIdToLastAccessTime) {
+      long currentLogicTime = mLogicTimeCount.incrementAndGet();
+      // update CRF value
+      // CRF(currentLogicTime)=CRF(lastAccessTime)*F(currentLogicTime-lastAccessTime)+F(0) 
+      if (mBlockIdToCRFValue.containsKey(blockId)) {
```

Comment:
line 296~304 is identical to line 314~322, I would suggest to consolidate them into a helper method.